File: backend/PasswordChecker.java
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;

public class PasswordChecker {
    public static void main(String[] args) {
        BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
        String encodedPassword = "$2a$10$shE2geCPkGP0SMx2yiY8C.HTTOOEm7j1RB8zj4AmPk5eD71B88YPe";
        String rawPassword = "202401";
        
        boolean matches = encoder.matches(rawPassword, encodedPassword);
        System.out.println("Password matches: " + matches);
    }
}

File: backend/password_checker.py
import bcrypt

# 从数据库中获取的加密密码
encoded_password = b'$2a$10$shE2geCPkGP0SMx2yiY8C.HTTOOEm7j1RB8zj4AmPk5eD71B88YPe'
# 要验证的原始密码
raw_password = b'202401'

# 验证密码
matches = bcrypt.checkpw(raw_password, encoded_password)
print(f"Password matches: {matches}")